tensorflow/python/framework/device_spec.py

```diff
@@ -47,38 +47,55 @@ def _as_device_str_or_none(device_type):
 @tf_export("DeviceSpec", v1=[])
 class DeviceSpecV2(object):
   """Represents a (possibly partial) specification for a TensorFlow device.
-
+  
   `DeviceSpec`s are used throughout TensorFlow to describe where state is stored
   and computations occur. Using `DeviceSpec` allows you to parse device spec
   strings to verify their validity, merge them or compose them programmatically.
-
+  
   Example:
-
+  
   ```python
   # Place the operations on device "GPU:0" in the "ps" job.
   device_spec = DeviceSpec(job="ps", device_type="GPU", device_index=0)
-  with tf.device(device_spec):
+  with tf.device(device_spec.to_string()):
     # Both my_var and squared_var will be placed on /job:ps/device:GPU:0.
     my_var = tf.Variable(..., name="my_variable")
     squared_var = tf.square(my_var)
   ```
-
+  
+  With eager execution disabled (by default in TensorFlow 1.x and by calling
+  disable_eager_execution() in TensorFlow 2.x), the following syntax
+  can be used:
+  
+  ```python
+  # Location in TensorFlow 2.x
+  from tensorflow.python.framework.ops import disable_eager_execution
```

Comment:
`tf.compat.v1.disable_eager_execution` is the recommended API.

Reply:
Yeah I saw that. But I thought it would be better to avoid the v1 API. Will update it right away.